File: Java/src/test/java/org/revian/MyLinkedListTest.java
package org.revian;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MyLinkedListTest {
    private MyLinkedList linkedList;

    @BeforeEach
    void setUp() {
        linkedList = new MyLinkedList();
    }

    @Test
    void testAddAndGet() {
        linkedList.add(1);
        linkedList.add(2);
        assertEquals(1, linkedList.get(0));
        assertEquals(2, linkedList.get(1));
    }

    @Test
    void testSize() {
        linkedList.add(1);
        linkedList.add(2);
        assertEquals(2, linkedList.size());
    }

    @Test
    void testGetInvalidIndex() {
        linkedList.add(1);
        assertThrows(IndexOutOfBoundsException.class, () -> linkedList.get(2));
    }
}

File: Scala/src/test/scala/org/revian/MyLinkedListTest.scala
package org.revian

import org.scalatest.funsuite.AnyFunSuite

class MyLinkedListTest extends AnyFunSuite {

  test("Add element to LinkedList") {
    val linkedList = new MyLinkedList
    linkedList.add(10)
    assert(linkedList.size() == 1)
    assert(linkedList.get(0) == 10)
  }

  test("Get element by index") {
    val linkedList = new MyLinkedList
    linkedList.add(10)
    linkedList.add(20)
    assert(linkedList.get(0) == 20)
    assert(linkedList.get(1) == 10)
  }

  test("Get element out of bounds should throw exception") {
    val linkedList = new MyLinkedList
    linkedList.add(10)
    linkedList.add(20)
    assertThrows[IndexOutOfBoundsException] {
      linkedList.get(3)
    }
  }

  test("Get from empty LinkedList should throw exception") {
    val linkedList = new MyLinkedList
    assertThrows[IndexOutOfBoundsException] {
      linkedList.get(0)
    }
  }

  test("Size of LinkedList should be correct") {
    val linkedList = new MyLinkedList
    linkedList.add(10)
    linkedList.add(20)
    linkedList.add(30)
    assert(linkedList.size() == 3)
  }
}
